src/components/layout/CardHeader.tsx

```diff
@@ -45,6 +46,10 @@ const CardHeader = function CardHeader({
 
   ...htmlAttributes
 }: CardHeaderProps) {
+  const closeableContext = useContext(CloseableContext);
+  // Provided `onClose` is prioritized, but also check to see if there is
+  // a close handler provided by a `CloseableContext`
+  const closeHandler = onClose ?? closeableContext?.onClose;
```

Comment:
Considering we are now adding this logic in more than one place, and we could even have more in the future, maybe we can have a custom hook which gets the main optional `onClose`, and internally checks for the context as well.

Something in these lines.

```ts
function useOnClose(onClose?: () => void) {
  const closeableContext = useContext(CloseableContext);
  // Provided `onClose` is prioritized, but also check to see if there is
  // a close handler provided by a `CloseableContext`
  return onClose ?? closeableContext?.onClose;
}
```

And then it's used like this:

```ts
const closeHandler = useOnClose(onClose);
```

Maybe it's a bit overkill, and it also has the problem that it has a less obvious purpose. On the other hand, it would avoid explaining/commenting every time why we do that, while also avoiding potential typos in future uses where we could do `closeableContext?.onClose ?? onClose;` by mistake.

I'll leave this up to your judgement.

Reply:
I think this is worth considering. Do you think we could perhaps extend on this current work (i.e. later) to add that hook? I already have a TODO to track some follow-up, because it's likely we'd want to make `Panel` "closeable" (not just `Dialog`).

Reply:
Yep, absolutely.